activerecord/test/cases/finder_test.rb

```diff
@@ -572,6 +572,20 @@ def test_take_and_first_and_last_with_integer_should_use_sql_limit
     assert_sql(/LIMIT|ROWNUM <=/) { Topic.last(5).entries }
   end
 
+  def test_last_with_offset
+    topic = Topic.order(:id)
+    assert_equal topic.to_a.last, topic.last
+    assert_equal topic.offset(1).to_a.last, topic.offset(1).last
+    assert_equal topic.last, topic.offset(4).last
+  end
```

Comment:
Both assertions should hold. Definitely `topic.offset(4).to_a.last` should give `topic.offset(4).last`, as you say. If you’d like, I can definitely add that assertion to the PR.

My motivation here was to have at least one test to capture the intuition that “last with offset” should give the same thing as “last without offset.” Because that idea wasn’t in the first two tests explicitly, I added this third one to make it more explicit.

That said, if you’d prefer the `topic.offset(4).to_a.last` assertion in addition to (or instead of) this one with `topic.last`, let me know and I’m more than happy to make that change.